bt-core/src/main/java/bt/data/file/FileSystemStorageUnit.java

```diff
@@ -175,24 +180,28 @@ public long capacity() {
 
     @Override
     public long size() {
-        return file.length();
```

Comment:
`File#length()` returns `0` for non-existent files, whereas `Files.size()` throws an `FileNotFoundException`.
This is causing the unit tests to fail. 
You could catch the exception and return 0, but it's probably more performant to just check `Files.exists()`  before reading the size.